src/box/xlog.h

```diff
@@ -215,7 +241,17 @@ static inline bool
 xdir_has_garbage(struct xdir *dir, int64_t signature)
 {
 	struct vclock *vclock = vclockset_first(&dir->index);
-	return vclock != NULL && vclock_sum(vclock) < signature;
+	if (vclock == NULL)
+		return false;
+
+	struct vclock retention;
+	xdir_get_retention_vclock(dir, &retention);
+	if (!vclock_is_set(&retention) ||
+	    vclock_compare(&retention, vclock) < 0)
```

Comment:
I don't get why shouldn't it be `vclock_compare(vclock, &retention) < 0`?

Reply:
This is an error, the test was also broken: I didn't initialize `expiration_time` of vclocks, which were before the retention. So, `retention_index_get` always returned uninitialized vclock, check was always true. Last `false` was never returned. 

Fixed this and the test in EE

Reply:
¡Gracias!